internal/store/builder.go

```diff
@@ -329,10 +331,9 @@ func (b *Builder) reflectorPerNamespace(
 	store cache.Store,
 	listWatchFunc func(kubeClient clientset.Interface, ns string) cache.ListerWatcher,
 ) {
-	for _, ns := range b.namespaces {
-		lw := listWatchFunc(b.kubeClient, ns)
-		instrumentedListWatch := watch.NewInstrumentedListerWatcher(lw, b.metrics, reflect.TypeOf(expectedType).String())
-		reflector := cache.NewReflector(sharding.NewShardedListWatch(b.shard, b.totalShards, instrumentedListWatch), expectedType, store, 0)
-		go reflector.Run(b.ctx.Done())
-	}
+	lwf := func(ns string) cache.ListerWatcher { return listWatchFunc(b.kubeClient, ns) }
+	lw := listwatch.MultiNamespaceListerWatcher(log.NewNopLogger(), b.namespaces, nil, lwf)
```

Comment:
I am not too sure about injecting a Noop logger. There definitely is value in logging the ops here. This goes back to my first comment.